test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/schema.py

```diff
@@ -542,7 +549,7 @@ def list_dict_option_schema(for_collection, plugin_type):
     basic_option_schema = {
         Required('description'): doc_string_or_strings,
         'required': bool,
-        'choices': list,
+        'choices': Any(list, dict),
```

Comment:
I would really use
```suggestion
        'choices': Any(list, {json_value: doc_string_or_strings}),
```
here, otherwise there's nothing that makes sure that the values of the dict are strings or lists of strings.

Reply:
hmm, this actually breaks validation as 'unhashable'

Reply:
How about using `object` instead of `json_value`? Or is that also not hashable?

Reply:
did just that, now on new error